neuralpp/test/quick_tests/symoblic/expression_test.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+
+from neuralpp.symbolic.function import Add
+from neuralpp.symbolic.basic_expression import BasicFunctionApplication, BasicConstant, BasicVariable
+    
+
+def test_basic_constant():
+    # Constant can be anything
+    constant_one = BasicConstant(1)
+    constant_abc = BasicConstant("abc")
+    assert constant_one != constant_abc
+    assert constant_one == BasicConstant(2-1)
+    assert constant_one.subexpression() == []
+    assert not constant_one.contains(constant_abc)
+    assert not constant_one.contains(constant_one)  # A constant does not contains() itself
+    assert constant_one.replace(constant_one, constant_abc) == constant_one  # Any replace() has no effect on constant
```

Comment:
Likewise, I think replace should replace constants too. If I say: "replace every occurrence of 1 by 2 in the following expression: 1", I think the answer should be 2, no?